crates/uv/tests/pip_install.rs

```diff
@@ -802,6 +812,41 @@ fn install_no_index_version() {
     context.assert_command("import flask").failure();
 }
 
+/// Install a package via --extra-index-url.
+///
+/// This is a regression test where previously `uv` would consult PyPI first,
+/// and if the package was found there, `uv` would not look at any other
+/// indexes. We fixed this by flipping the priority order of indexes so that
+/// PyPI becomes the fallback and the extra indexes are checked first.
+///
+/// (Neither approach matches `pip`'s behavior, which considers versions of
+/// each package from all indexes. `uv` stops at the first index it finds a
+/// package in.)
+///
+/// Ref: <https://github.com/astral-sh/uv/issues/1600>
+#[test]
+fn install_extra_index_url_has_priority() {
+    let context = TestContext::new("3.12");
+
+    uv_snapshot!(command_without_exclude_newer(&context)
+        .arg("--extra-index-url")
+        .arg("https://pypi.ngc.nvidia.com")
```

Comment:
I feel a little bad about doing this. I'm happy to try a different approach if folks have ideas.

It feels like maybe the better longer term solution is to add multi-index support to packse? (I briefly looked to see if it had it already, and it doesn't look like it does.)

Reply:
Thanks for the nudge. I managed to find an example that worked with test.pypi.org.